code/modules/paperwork/pen.dm

```diff
@@ -31,6 +31,32 @@
 	var/requires_gravity = TRUE // can you use this to write in zero-g
 	embedding = list(embed_chance = 50)
 	sharpness = SHARP_POINTY
+	var/dart_insert_icon = 'icons/obj/weapons/guns/toy.dmi'
+	var/dart_insert_casing_icon_state = "overlay_pen"
+	var/dart_insert_projectile_icon_state = "overlay_pen_proj"
+
+/obj/item/pen/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/dart_insert, dart_insert_icon, dart_insert_casing_icon_state, dart_insert_icon, dart_insert_projectile_icon_state)
+	RegisterSignal(src, COMSIG_DART_INSERT_ADDED, PROC_REF(on_inserted_into_dart))
+	RegisterSignal(src, COMSIG_DART_INSERT_REMOVED, PROC_REF(on_removed_from_dart))
+	RegisterSignal(src, COMSIG_DART_INSERT_GET_VAR_MODIFIERS, PROC_REF(get_dart_var_modifiers))
+
+/obj/item/pen/proc/on_inserted_into_dart(datum/source, obj/projectile/dart, mob/user, embedded = FALSE)
+	SIGNAL_HANDLER
+
+/obj/item/pen/proc/get_dart_var_modifiers(datum/source, list/modifiers)
+	SIGNAL_HANDLER
+	modifiers["damage"] = max(5, throwforce)
+	modifiers["speed"] = max(0, throw_speed - 3)
+	modifiers["embedding"] = embedding
+	modifiers["armour_penetration"] = armour_penetration
+	modifiers["wound_bonus"] = wound_bonus
+	modifiers["bare_wound_bonus"] = bare_wound_bonus
+	modifiers["demolition_mod"] = demolition_mod
```

Comment:
Why do we do this via signal, why don't we just pass it in as an argument to the component?

Reply:
mostly so that the admins can VV the properties of the pen at runtime and it affects how the pen affects the dart it gets put into. Unless you would suggest adding a signal handler like `COMSIG_VAR_EDIT` or something.

Reply:
So follow up question why don't we just grab modifiers straight in the proc itself? Would keep it up to date with the properties of the pen.

Reply:
I was intending to make it so that the correlation between projectile modifications and parent vars isn't hardcoded - particularly so that lua scripts can be used to define custom behavior.